backend/routes/nft.go

```diff
@@ -195,3 +206,143 @@ func mintNFTDevnet(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Write([]byte("Minted NFT on devnet"))
 }
+
+func LikeNFT(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var nftlikeReq NFTLikesRequest
+	err := json.NewDecoder(r.Body).Decode(&nftlikeReq)
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "error decoding JSON").Error(), http.StatusBadRequest)
+		return
+	}
+
+	log.Printf("Received request: %+v", nftlikeReq)
+
+	// Check if the user has already liked the NFT
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), "SELECT * FROM NFTLikes WHERE nftKey = $1 AND liker = $2", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
```

Comment:
See the comment on `init.sql` then you wont need to do this check.

Reply:
without this, am getting the following error

 `error inserting into database: ERROR: duplicate key value violates unique constraint "nftlikes_pkey"`

this is after i change the schema to this 

```
CREATE TABLE NFTLikes (
  nftKey integer NOT NULL,
  liker char(64) NOT NULL,
  time timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (nftKey, liker)
);
```

Reply:
Thus the check is need before INSERTION

Reply:
You're getting that error when trying to like something that is already liked?

If so that is good, then you can catch the error on the backend and return an error message to the user that the nft is already liked.

Reply:
The new commit contain exactly what you describe above

Reply:
This check can be removed, though, and then check for the uniqueness error in the INSERT command.

It's best not to do multiple postgres commands.